src/parse/mod.rs

```diff
@@ -326,7 +311,7 @@ impl<I: Iterator<Item = Lexeme>> Parser<I> {
     fn lex_error(&mut self, err: CompileError) {
         self.error_handler.push_back(err);
     }
-    pub fn collect_results(self) -> (Vec<Locatable<Declaration>>, Vec<CompileError>) {
+    pub fn collect_results(&mut self) -> (Vec<Locatable<Declaration>>, Vec<CompileError>) {
```

Comment:
Where do you need to use the parser after collecting the results from it?

Reply:
https://github.com/jyn514/rcc/pull/172/files#diff-b4aea3e418ccdb71239b96952d9cddb6R92

Reply:
If it's an empty program error isn't the location usually zero? Also for the empty program error I don't think it matters to the user where that error is, just the file